src/main/java/io/confluent/connect/hdfs/HdfsSinkTask.java

```diff
@@ -87,12 +87,24 @@ public void start(Map<String, String> props) {
     } catch (ConfigException e) {
       throw new ConnectException("Couldn't start HdfsSinkConnector due to configuration error.", e);
     } catch (ConnectException e) {
+      // Log at info level to help explain reason, but Connect logs the actual exception at ERROR
       log.info("Couldn't start HdfsSinkConnector:", e);
       log.info("Shutting down HdfsSinkConnector.");
       if (hdfsWriter != null) {
-        hdfsWriter.close();
-        hdfsWriter.stop();
+        try {
+          try {
+            log.debug("Closing data writer due to task start failure.");
+            hdfsWriter.close();
+          } finally {
+            log.debug("Stopping data writer due to task start failure.");
+            hdfsWriter.stop();
+          }
+        } catch (Throwable t) {
+          log.debug("Error closing and stopping data writer: {}", t.getMessage(), t);
+        }
       }
+      // Always throw the original exception that prevent us from starting
+      throw e;
```

Comment:
It seems really strange that the task caught but did not rethrow the ConnectException. (NPEs and other runtimetimes would not be caught and would be handled by Connect.) By catching but not rethrowing the exception, Connect does not know task startup failed and will continue using the task.

This makes me wonder if this was intentional, though the existing log message still says "Shutting down HdfsSinkConnector".

Reply:
great catch! I wonder if `ConnectException` is throw by any of the code below? this pattern of catching this exception and swallowing it seems to appear in multiple places. maybe that's why this went unnoticed?.

Reply:
It definitely happens: DataWriter constructor throws ConnectException with any problems.